src/NewRelic.Core/NewRelic.Collections/ConcurrentPriorityQueue.cs

```diff
@@ -122,13 +126,13 @@ private void RemoveItemsFromBottomOfSortedSet()
             if (Size == 0)
             {
                 Reset();
```

Comment:
If Size is 0, won't `Reset()` clear `_sortedSet`, meaning you won't increment the dropped counter?

Reply:
Good catch! I reworked the logic in `Reset()` to capture the size of the collection before clearing it.